src/test/java/org/apache/maven/shared/artifact/filter/GNPatternIncludesArtifactFilter.java

```diff
@@ -0,0 +1,917 @@
+                            emptyOrChars( artifact.getClassifier() ),
+                            emptyOrChars( artifact.getBaseVersion() )
+                    };
+        Boolean match = match( artifactGatvCharArray );
+        if ( match != null )
+        {
+            return match;
+        }
+
+        if ( actTransitively )
+        {
+            final List<String> depTrail = artifact.getDependencyTrail();
+
+            if ( depTrail != null && depTrail.size() > 1 )
+            {
+                for ( String trailItem : depTrail )
+                {
+                    char[][] depGatvCharArray = tokenizeAndSplit( trailItem );
+                    match = match( depGatvCharArray );
+                    if ( match != null )
+                    {
+                        return match;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private Boolean match( char[][] gatvCharArray )
+    {
+        for ( Pattern pattern : patterns )
+        {
+            if ( pattern.matches( gatvCharArray ) )
+            {
+                patternsTriggered.add( pattern );
+                return !( pattern instanceof NegativePattern );
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * <p>addFilteredArtifact.</p>
+     *
+     * @param artifact add artifact to the filtered artifacts list.
+     */
+    protected void addFilteredArtifact( final Artifact artifact )
+    {
+        filteredArtifact.add( artifact );
+    }
+
+    /** {@inheritDoc} */
+    public void reportMissedCriteria( final Logger logger )
+    {
+        // if there are no patterns, there is nothing to report.
+        if ( !patterns.isEmpty() )
+        {
+            final List<Pattern> missed = new ArrayList<>( patterns );
+            missed.removeAll( patternsTriggered );
+
+            if ( !missed.isEmpty() && logger.isWarnEnabled() )
+            {
+                final StringBuilder buffer = new StringBuilder();
+
+                buffer.append( "The following patterns were never triggered in this " );
+                buffer.append( getFilterDescription() );
+                buffer.append( ':' );
+
+                for ( Pattern pattern : missed )
+                {
+                    buffer.append( "\no  '" ).append( pattern ).append( "'" );
+                }
+
+                buffer.append( "\n" );
+
+                logger.warn( buffer.toString() );
+            }
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String toString()
+    {
+        return "Includes filter:" + getPatternsAsString();
+    }
+
+    /**
+     * <p>getPatternsAsString.</p>
+     *
+     * @return pattern as a string.
+     */
+    protected String getPatternsAsString()
+    {
+        final StringBuilder buffer = new StringBuilder();
+        for ( Pattern pattern : patterns )
+        {
+            buffer.append( "\no '" ).append( pattern ).append( "'" );
```

Comment:
`System.lineSeparator()`

Reply:
This class IS A COPY of old implementation moved to src/test and retained for perf comparison purposes only. Am not modifying anything in it, it should remain as it was on master before this PR